src/utils.js

```diff
@@ -36,6 +37,54 @@ function emitEvent(...args) {
   }
 }
 
+/* eslint-disable no-restricted-properties */
+export function getWinDimensions(path = null) {
+  if (!windowDimensions) {
+    const top = canAccessWindowTop() ? internal.getWindowTop() : internal.getWindowSelf();
+
+    windowDimensions = {
+      screen: {
+        width: top.screen?.width,
+        height: top.screen?.height,
+        availWidth: top.screen?.availWidth,
+        availHeight: top.screen?.availHeight,
+        colorDepth: top.screen?.colorDepth,
+      },
+      innerHeight: top.innerHeight,
+      innerWidth: top.innerWidth,
+      outerWidth: top.outerWidth,
+      outerHeight: top.outerHeight,
+      visualViewport: {
+        height: top.visualViewport?.height,
+        width: top.visualViewport?.width,
+      },
+      document: {
+        documentElement: {
+          clientWidth: top.document?.documentElement?.clientWidth,
+          clientHeight: top.document?.documentElement?.clientHeight,
+          scrollTop: top.document?.documentElement?.scrollTop,
+          scrollLeft: top.document?.documentElement?.scrollLeft,
+        },
+        body: {
+          scrollTop: document.body?.scrollTop,
+          scrollLeft: document.body?.scrollLeft,
+          clientWidth: document.body?.clientWidth,
+          clientHeight: document.body?.clientHeight,
+        },
+      }
+    };
+  };
+
+  return path ? deepAccess(windowDimensions, path) : windowDimensions;
+}
+/* eslint-enable no-restricted-properties */
+
+export function resetWinDimensions() {
+  windowDimensions = null;
+}
+
+window.addEventListener('resize', resetWinDimensions);
+
```

Comment:
shouldn't it also reset on scroll, as this would update the scroll properties?

Reply:
Would it be better to debounce, similar to https://github.com/prebid/Prebid.js/pull/12848? so that we don't risk missing some case where the dimensions change. 

Maybe refactor both cases to debounce for a short time (say 20ms). Compared to the interval between auctions, it would behave better if this is used somewhere other than building request - for example, in bid won handlers and so on.